### PR TITLE
Add promise queueing to try and make batch operations faster

### PR DESCRIPTION
**Pull request by AWare**
## What does this change?

This will only allow two batch operations or image uploads to run concurrently. I've used this pattern a few times to fix similar problems so am hopeful it'll help things.

## How can success be measured?

Increased performance when running batch jobs.

## Who should look at this?
@akash1810 @paperboyo @mbarton 
Could also do with a hand spinning up grid locally or on test to test this. 

## Tested?
- [ ] locally
- [ ] on TEST

### CODE REVIEW

**Comment by paperboyo** [comment]
I’ll let Akash and Micheal have proper opinions, but just a quation from a village idiot: which batch operations does this apply to? Uploads? Metadata editing? Deletions? Only through UI or mediaAPI too?

**Comment by AWare** [comment]
Just in the UI, and afaict, all of the above as I changed the helper that appears used in all of those places. JS is very good at queuing a far larger number of promises concurrently than it's able to deliver on.

**Comment by paperboyo** [comment]
Looking forward to test batch operations on TEST, then 👏 .

File: kahuna/public/js/upload/jobs/upload-jobs.js [hunk 1]
```diff
@@ -1,178 +1,196 @@
...
+          error => {
             const reason = error.body && error.body.errorKey;
 
-            const message = reason === 'unsupported-type' ?
-                'The Grid only supports JPG and PNG images.' +
-                ' Please convert the image and try again.' :
-                error.body && error.body.errorMessage || 'unknown';
+            const message =
+              reason === "unsupported-type"
+                ? "The Grid only supports JPG and PNG images." +
```
Comment (mbarton) [inline kahuna/public/js/upload/jobs/upload-jobs.js]: Unrelated but does this message need changing now TIFF support has been added? cc @paperboyo

Comment (paperboyo) [reply kahuna/public/js/upload/jobs/upload-jobs.js]: Thanks for review, @mbarton! Yeah, also seen it, but TIFF support hasn’t been advertised yet. Reasoning (not too strong, I admit) being outstanding [work in this](https://github.com/guardian/grid/issues/2395).

**Comment by AWare** [comment]
@mbarton is there a solution to the linter noise?

**Pull request closed without merging**